src/observers/link_prefetch_observer.js

```diff
@@ -77,12 +86,18 @@ export class LinkPrefetchObserver {
         )
 
         prefetchCache.setLater(location.toString(), fetchRequest, this.#cacheTtl)
-
-        link.addEventListener("mouseleave", () => prefetchCache.clear(), { once: true })
```

Comment:
@afcapel this line is the crux of this PR. Adding listeners directly to the `<a>` without removing them poses a risk of leaking. Additionally, there isn't a parallel pattern for `touchstart` (and `touchend`) events.

The rest of the changes in this diff support that setup-and-teardown.

Once this is merged, adding support for [focusin](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event) and [focusout](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event) events would be fairly trivial.

Reply:
@afcapel if this is going to be enabled by default, I think this change is important to consider for touch devices.

Reply:
>Adding listeners directly to the < a > without removing them poses a risk of leaking memory from the instances.

When `once` is `true` [the listener would be automatically removed when invoked](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once). So there wouldn't be a risk of memory leaks. Or am I missing something?

Reply:
Ahh just read the description, got it. Sorry, was linked directly to this comment from https://github.com/hotwired/turbo/pull/1167 😃